### ISSUE TITLE
Redirection router priority only works if it is defined

### ISSUE THREAD

**Issue by Totalus** [traefik/traefik#9027]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

The [redirection](https://doc.traefik.io/traefik/routing/entrypoints/#redirection) configuration creates a router with a default priority of 1, as per the documentation.

I created a second router with a higher priority.

I was expecting this second router to handle my request, prevailing on the redirection because of its higher priority.

### What did you see instead?

The redirection router handles the request, unless the priority of the redirection is specifically defined.

### What version of Traefik are you using?

Tried on `v2.6` and `v2.7` docker images.

### What is your environment & configuration?

<details><summary>Here's a docker-compose file to reproduce.</summary>

```yaml
version: "3.7"

services:
  traefik:
    image: traefik:v2.6
    container_name: traefik
    command:
      - "--log.level=debug"
      - "--api.dashboard=true"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=true"
      - "--providers.docker.network=web"
      # HTTP endpoint
      - "--entrypoints.web.address=:80"
      # Redirect HTTP to HTTPS
      - "--entrypoints.web.http.redirections.entryPoint.to=websecure"
      - "--entrypoints.web.http.redirections.entryPoint.scheme=https"
      - "--entrypoints.web.http.redirections.entryPoint.permanent=true"
      - "--entrypoints.web.http.redirections.entryPoint.priority=1"
      # HTTPS endpoint
      - "--entrypoints.websecure.address=:443"
      - "--entrypoints.websecure.http.tls={}"
    ports:
      - "80:80"
      - "443:443"
      - "127.0.0.1:8080:8080"
    volumes:
      - //var/run/docker.sock://var/run/docker.sock

  hello:
    image: nginxdemos/hello
    container_name: hello
    labels:
        # Use web endpoint
        - traefik.enable=true
        - traefik.http.routers.hello2.rule=Host(`localhost`) && Path(`/hello`)
        - traefik.http.routers.hello2.entrypoints=web
        - traefik.http.routers.hello2.priority=2000

networks:
  default:
    name: "web"

```
</details>

The `--entrypoints.web.http.redirections.entryPoint.priority=1` argument defines the redirection router priority.

- When the redirection priority is not defined, the redirection router handles the request.
- When it is defined, the request is properly routed to the hello container.

**Comment by ldez** [comment]
Hello,

The entrypoint redirection configuration creates a router with a default priority of `MaxInt32 - 1`

The higher the value, the higher the priority of the router.

So if you want to create a router prevailing to the redirection, your router must have a priority higher than `2147483646` (ex: `2147483650`)

Or you can set a value than your router in the entrypoint configuration (`entrypoints.foo.http.redirections.entrypoint.priority`)

https://doc.traefik.io/traefik/routing/entrypoints/#entrypoint

The documentation is wrong about the default priority value, I will fix that.

**Comment by tomMoulard** [comment]
Fixed by #9028
